### ISSUE TITLE
Conditionally rendered element under fragment in island does not render

### ISSUE THREAD

**Issue by crowlKats** [denoland/fresh#502]
```tsx
/** @jsx h */
/** @jsxFrag Fragment */
import { h, Fragment } from "preact";
import { useState } from "preact/hooks";

export default function Component() {
  const [show, setShow] = useState(false);

  return (
    <>
      Hello
      <div onClick={() => setShow(true)}>
        World
      </div>
      {show && (
        <div>I'm not rendered</div>
      )}
    </>
  );
}
```
This will error with `Uncaught (in promise) DOMException: Node.insertBefore: Child to insert before is not a child of this node` on the client when `World` is clicked.

**Comment by m4rocks** [comment]
I also ran into this issue. Rendering an empty div instead should fix the problem

**Comment by psimk** [comment]
Based on the error message, the error occurs in [`createRootFragment`](https://github.com/denoland/fresh/blob/0cb05d1dac13d14b6b31d0e61d2e6f0ccfb5d4a4/src/runtime/main.ts#L4). Specifically when preact calls `appenChild`, which subsequently calls our [`insert`](https://github.com/denoland/fresh/blob/0cb05d1dac13d14b6b31d0e61d2e6f0ccfb5d4a4/src/runtime/main.ts#L10) function.

I think part of the issue is related to the way we instantiate the [`s`](https://github.com/denoland/fresh/blob/0cb05d1dac13d14b6b31d0e61d2e6f0ccfb5d4a4/src/runtime/main.ts#L9) variable: it is assigned to the `nextSibling` of the last `replaceNode`. In other words, [`s`](https://github.com/denoland/fresh/blob/0cb05d1dac13d14b6b31d0e61d2e6f0ccfb5d4a4/src/runtime/main.ts#L9) is assigned to the next sibling of the final child of our island component. This happens to be the end tag node, that we use to denote the end of the island component.

This does not seem intentional, as right after we create the root fragment we [remove the end tag](https://github.com/denoland/fresh/blob/0cb05d1dac13d14b6b31d0e61d2e6f0ccfb5d4a4/src/runtime/main.ts#L58). Which means that [`s`](https://github.com/denoland/fresh/blob/0cb05d1dac13d14b6b31d0e61d2e6f0ccfb5d4a4/src/runtime/main.ts#L9) will always point to a comment element which doesn't exist in the DOM.

---

Moving the instantiation of [`s`](https://github.com/denoland/fresh/blob/0cb05d1dac13d14b6b31d0e61d2e6f0ccfb5d4a4/src/runtime/main.ts#L9) into [`insert`](https://github.com/denoland/fresh/blob/0cb05d1dac13d14b6b31d0e61d2e6f0ccfb5d4a4/src/runtime/main.ts#L10) fixes the error ([psimk/fresh/#bug/root-fragment-in-insland](https://github.com/psimk/fresh/tree/bug/root-fragment-in-insland)), and @crowlKats's example runs as expected. However, I am not sure why, but preact always calls `appendChild`. Which results in the conditional part of the component to always be appended to the end.

If we swap the location of the "show" portion of the above example, and run it with the moved [`s`](https://github.com/denoland/fresh/blob/0cb05d1dac13d14b6b31d0e61d2e6f0ccfb5d4a4/src/runtime/main.ts#L9) instantiation. The "show" portion will be rendered **after** `<div>World</div>`.

 ```tsx
 /** @jsx h */
 /** @jsxFrag Fragment */
 import { h, Fragment } from "preact";
 import { useState } from "preact/hooks";
 
 export default function Component() {
   const [show, setShow] = useState(false);
 
   return (
     <>
       {show && (
         <div>I'm not rendered where I should be rendered</div>
       )}
       Hello
       <div onClick={() => setShow(true)}>
         World
       </div>
     </>
   );
 }
 ```

**Comment by TranquilMarmot** [comment]
I also ran into this with a "code only" island

For example,

```tsx
const SomeIsland = () => {
   useEffect(() => {
      console.log("Island in the sun");
   }, []);

  return null; // or return <></>
}
```

Results in:

> Uncaught TypeError: Cannot read properties of undefined (reading 'nextSibling') at createRootFragment

Returning an empty `<div />` here works, but it would be nice to not have to pollute the DOM when you just want to run some client-side JS.

**Comment by TranquilMarmot** [comment]
This bit me again in the same night while trying to use the [menu component from `@headless-ui/react`](https://headlessui.com/react/menu)! See this discussion: https://github.com/denoland/fresh/discussions/606#discussioncomment-3902802

**Comment by rimzudev** [comment]
Is there an update or workaround on this? I think I am having a slightly different case with the same problem in `fresh@1.1.4`. Minimalistic example:
```js
// islands/Show.tsx
import { useState } from "https://esm.sh/preact@10.11.0/hooks";

export default function ShowDiv() {
  const [open, setOpen] = useState(true);

  return (
    <>
      <button onClick={() => setOpen(true)}>
        Open
      </button>
      {open
        ? (
          <div>
            <button
              type="button"
              onClick={() => setOpen(false)}
            >
              Close
            </button>
          </div>
        )
        : null}
    </>
  );
}
```
I want the div to exist conditionally. Closing it the first time works but opening it in any way always results in:
```
Uncaught DOMException: Failed to execute 'insertBefore' on 'Node': The node before which the new node is to be inserted is not a child of this node.
    at Object.insert ...
```

**Comment by rimzudev** [comment]
I just realized, it works by turning the empty tag (`<>`) into a div (`<div>`):
```js
// ...
  return (
    <div>
      <button onClick={() => setOpen(true)}>
        Open
      </button>
      {open
        ? (
          <div>
            <button
              type="button"
              onClick={() => setOpen(false)}
            >
              Close
            </button>
          </div>
        )
        : null}
    </div>
  );
}
```
_I hope this helps some people with the same problem or by solving this in general as having the button placed right there without div enclosing it would help me. But maybe this is not even intended. I'll leave the explanation why this works/does not work to someone else. 👀_

**Comment by azerum** [comment]
Am I missing something, or `insertBefore()` can be implemented as it is - by calling `parent.insertBefore()`?

```typescript
function createRootFragment(
  parent: Element,
  replaceNode: Node | Node[],
) {
  replaceNode = ([] as Node[]).concat(replaceNode);

  // @ts-ignore this is fine
  return parent.__k = {
    nodeType: 1,

    parentNode: parent,

    firstChild: replaceNode[0],
    childNodes: replaceNode,

    insertBefore(node: Node, child: Node) {
      parent.insertBefore(node, child);
    },

    appendChild(child: Node) {
      parent.appendChild(child);
    },

    removeChild(child: Node) {
      parent.removeChild(child);
    },
  };
}
```

I've forked the repository, added this modifications, and all tests run. I've also added test cases for two example comments above. Maybe tests don't tell a complete picture though. See #1172

**Comment by iuioiua** [comment]
This has been fixed in #1172.
